src/main/java/org/candlepin/subscriptions/tally/InventorySwatchDataCollator.java

```diff
@@ -248,29 +258,31 @@ public static SortKey fromHbiSystem(InventoryHostFacts system) {
           .hardwareSubmanId(hardwareSubmanId)
           .hypervisorUuid(hypervisorUuid)
           .inventoryId(system.getInventoryId().toString())
+          .instanceId(system.getProviderId())
           .build();
     }
 
     @Override
     public int compareTo(@NotNull SortKey other) {
-      var hardwareSubmanIdResult =
-          Objects.compare(
-              hardwareSubmanId,
-              other.getHardwareSubmanId(),
-              Comparator.nullsLast(Comparator.naturalOrder()));
+      var hardwareSubmanIdResult = compare(hardwareSubmanId, other.getHardwareSubmanId());
       if (hardwareSubmanIdResult != 0) {
         return hardwareSubmanIdResult;
       }
-      var hypervisorUuidResult =
-          Objects.compare(
-              hypervisorUuid,
-              other.getHypervisorUuid(),
-              Comparator.nullsLast(Comparator.naturalOrder()));
+      var hypervisorUuidResult = compare(hypervisorUuid, other.getHypervisorUuid());
       if (hypervisorUuidResult != 0) {
         return hypervisorUuidResult;
       }
-      return Objects.compare(
-          inventoryId, other.getInventoryId(), Comparator.nullsLast(Comparator.naturalOrder()));
+
+      var instanceIdResult = compare(instanceId, other.getInstanceId());
+      if (instanceIdResult == 0) {
+        return 0;
+      }
```

Comment:
I don't think this is super necessary, but I've changed it.

Reply:
Actually, this is critical because it affects whether the HBI system stream or swatch system stream is advanced when they are different systems. The collation breaks if the comparison logic is not the same as the database sorting.